src/stores/QuestionOfDayStore.ts

```diff
@@ -0,0 +1,39 @@
+import { derive } from 'valtio/utils'
+import { proxy } from 'valtio'
+import getPostStatuses from 'helpers/getPostStatuses'
+import getPostStorage from 'helpers/getPostStorage'
+import getQuestionOfTheDayIds from 'helpers/getQuestionOfTheDayIds'
+import safeTransformPostOutput from 'helpers/safeTransformPostOutput'
+
+interface QodStoreType {
+  qodAddress: Promise<string>
+  allQodPostIds: Promise<number[]>
+}
+
+const contract = getPostStorage()
+
+const state = proxy<QodStoreType>({
+  qodAddress: contract.replyAllAddress(),
+  allQodPostIds: getQuestionOfTheDayIds(contract),
+})
+
+const QodStore = derive(
+  {
+    lastQoDPost: async (get) => {
+      const qodIds = await get(state).allQodPostIds
+      const lastId = qodIds.length ? Math.max(...qodIds) : undefined
```

Comment:
If you do:
`const lastId = qodIds.length && Math.max(...qodIds)`, it will return `false | id`, so `if` statement will work anyway

Reply:
Agreed, thx 🙂